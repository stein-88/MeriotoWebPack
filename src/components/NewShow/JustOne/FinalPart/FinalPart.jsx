import React from 'react'
// import { v4 as uuidv4 } from 'uuid'
import schema from './data'

const FinalPart = ({ ...props }) => {
    const { novoloop } = schema
    return (
        <div className="col-4">
            {novoloop && novoloop.length > 0
                && novoloop.map((ele, ind) => {
                    if (!ele) return null
                    let finalValue = props[ele.refer]
                    if (ele.yesORno) finalValue = props[ele.refer] ? 'yes' : 'no'
                    if (ele.special) finalValue = <a href={props[ele.refer]} target="_blank" rel="noreferrer">{ele.schema}</a>
                    if (!finalValue) return null
                    return (
                        <p
                            key={`keynovolooping${ind + 1}`}
                            className="mb-0"
                        >
                            <small>
                                <strong data-test="12elements">
                                    {ele.schema === 'API_URL' ? finalValue : ele.schema}
                                    {' '}
                                </strong>
                                {ele.schema !== 'API_URL' && finalValue}
                            </small>
                        </p>
                    )
                })}
        </div>
    )
}

export default FinalPart
