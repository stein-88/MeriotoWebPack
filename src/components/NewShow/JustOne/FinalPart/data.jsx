const schemafinalpart = {
    novoloop: [
        {
            refer: 'units',
            schema: 'Units:',
        },
        {
            refer: 'unit_qty',
            schema: 'Unit qty:',
        },
        {
            refer: 'qty',
            schema: 'qty:',
        },
        {
            refer: 'for_company',
            schema: 'For company:',
            yesORno: true,
        },
        {
            refer: 'for_charity',
            schema: 'For charity:',
            yesORno: true,
        },
        {
            refer: 'for_individual',
            schema: 'For individual:',
            yesORno: true,
        },
        {
            refer: 'for_internal',
            schema: 'For internal:',
            yesORno: true,
        },
        {
            refer: 'is_license_required',
            schema: 'License is required:',
            yesORno: true,
        },
        {
            refer: 'reason_for_listing',
            schema: 'Reason for listing:',
        },
        {
            refer: 'id',
            schema: 'ID:',
        },
        {
            refer: 'created',
            schema: 'Created:',
        },
        {
            refer: 'url',
            schema: 'API_URL',
            special: true,
        },
    ],
}

export default schemafinalpart
