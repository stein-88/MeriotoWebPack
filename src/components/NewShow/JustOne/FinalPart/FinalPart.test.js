import React from 'react'
import renderer from 'react-test-renderer'
import FinalPart from './FinalPart'

describe('Test on FinalPart Component', () => {
    const compFinalPart = renderer.create(<FinalPart />)
    test('Sould render with no props', () => {
        const jsonFinalPart = compFinalPart.toJSON()
        expect(jsonFinalPart).toMatchSnapshot()
    })
   /*  test('Sould have 12 elements', () => {
        const compFinalPart = renderer.create(<FinalPart />)
        const Result = compFinalPart.root.findAllByProps({ ['data-test']: '12elements' })
        expect(Result).toHaveLength(12)
    }) */
})