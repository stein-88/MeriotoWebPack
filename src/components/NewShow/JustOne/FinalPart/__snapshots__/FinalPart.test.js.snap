// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on FinalPart Component Sould render with no props 1`] = `
<div
  className="col-4"
>
  <p
    className="mb-0"
  >
    <small>
      <strong
        data-test="12elements"
      >
        For company:
         
      </strong>
      no
    </small>
  </p>
  <p
    className="mb-0"
  >
    <small>
      <strong
        data-test="12elements"
      >
        For charity:
         
      </strong>
      no
    </small>
  </p>
  <p
    className="mb-0"
  >
    <small>
      <strong
        data-test="12elements"
      >
        For individual:
         
      </strong>
      no
    </small>
  </p>
  <p
    className="mb-0"
  >
    <small>
      <strong
        data-test="12elements"
      >
        For internal:
         
      </strong>
      no
    </small>
  </p>
  <p
    className="mb-0"
  >
    <small>
      <strong
        data-test="12elements"
      >
        License is required:
         
      </strong>
      no
    </small>
  </p>
  <p
    className="mb-0"
  >
    <small>
      <strong
        data-test="12elements"
      >
        <a
          rel="noreferrer"
          target="_blank"
        >
          API_URL
        </a>
         
      </strong>
    </small>
  </p>
</div>
`;
